x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -281,6 +279,7 @@
                         new RoleDescriptor.IndicesPrivileges[] {
                                 RoleDescriptor.IndicesPrivileges.builder()
                                         .indices(".ml-anomalies*", ".ml-notifications*", ".ml-state*", ".ml-meta*", ".ml-stats-*")
+                                        .allowRestrictedIndices(true)
```

Comment:
I don't think `machine_learning_admin` should need to access system indices.  The `.ml` indices they are given access to (in the line above) are all hidden indices.  The idea is that ML APIs are used to manipulate the documents in system indices.

If there's a test that fails when `machine_learning_admin` cannot access restricted indices then let me know, because that's probably a bug that needs fixing.

Reply:
> The idea is that ML APIs are used to manipulate the documents in system indices.

And to complete the picture, when the ML APIs access the system indices they should be doing that as the internal `_xpack` user, which _does_ now need restricted indices allowed.

Reply:
Good catch @droberts195!

Looks like only `.ml-meta*`, `.ml-config*` and `.ml-inference-*` (yes, only the .ml-inference pattern has a dash appended) are registered as system indices, hence the above role change is not needed.
Indeed if this is added to pass a test, it is probably a bug.